benchmarking/src/performance_evaluation.py

```diff
@@ -0,0 +1,860 @@
+import abc
+import json
+import os
+import random
+import re
+import threading
+import time
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import pandas as pd
+from tqdm import tqdm
+
+from llmperf import common_metrics
+from llmperf.sambanova_client import llm_request
+from llmperf.models import RequestConfig
+from llmperf.utils import LLMPerfResults, flatten, get_tokenizer
+
+MODEL_TYPE_IDENTIFIER = {"mistral": "mistral", "llama3": "llama-3"}
+PROMPT_GENERATION_OFFSET = 3
+
+class BasePerformanceEvaluator(abc.ABC):
+    def __init__(
+            self,
+            model_name: str,
```

Comment:
These inputs are consistent across both custom and synthetic datasets. Consequently, I have defined them in the base class so they will be inherited across both sub-classes.